### PR TITLE
chore: Minor cleanup `simplify_demo()` example

### PR DESCRIPTION
**Pull request by kavirajk**
## Which issue does this PR close?

Closes NA

## What changes are included in this PR?

fix comment in the `simplify_demo()` example to sync with code

## Are these changes tested?

Not needed. Jut fixes the comments.

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion-examples/examples/expr_api.rs [hunk 1]
```diff
@@ -209,7 +205,7 @@ fn simplify_demo() -> Result<()> {
     );
 
     // String --> Date simplification
-    // `cast('2020-09-01' as date)` --> 18500
+    // `cast('2020-09-01' as date)` --> 18506 # number of days since epoch 1970-01-01
```
Comment (kavirajk) [inline datafusion-examples/examples/expr_api.rs:208]:
```bash
-bash-5.2$ echo "scale=6; ($(gdate +%s -d "2020-09-01") - $(gdate +%s -d "1970-01-01")) / 86400" | bc
18505.958333
```

File: datafusion-examples/examples/expr_api.rs [hunk 2]
```diff
@@ -177,16 +177,12 @@ fn simplify_demo() -> Result<()> {
     );
 
     // here are some other examples of what DataFusion is capable of
-    let schema = Schema::new(vec![
-        make_field("i", DataType::Int64),
-        make_field("b", DataType::Boolean),
-    ])
-    .to_dfschema_ref()?;
+    let schema = Schema::new(vec![make_field("i", DataType::Int64)]).to_dfschema_ref()?;
```
Comment (kavirajk) [inline datafusion-examples/examples/expr_api.rs:180]: field `b` is not used

File: datafusion-examples/examples/expr_api.rs (hunk 1 above)
Comment (kavirajk) [reply datafusion-examples/examples/expr_api.rs:208]: Also matches with assert in the line below

**Review by alamb — APPROVED** [review]
Thank you @kavirajk

**Pull request merged**